### PR TITLE
3811 Fix displayCondition defined on object property inside array does not evaluate

### PR DESCRIPTION
**Pull request by ivicac**
- **3811 Handle remaining `[index]` placeholders with the last index value**
- **3811 Refactor `fieldType` references to use `inputs[index]` format for consistency**
- **3811 Reorganize**
- **3811 Add tests**
- **3811 Handle `[index]` placeholders in `displayCondition` by adding array evaluation logic**
- **3811 Handle edge case for `[index]` placeholders when `currentIndexes` is not empty**
- **3811 client - Handle displayCondition in `useObjectProperty` to ensure proper property inclusion**
- **3811 client - Add tests for display condition index replacement logic in `useProperty`**

### CODE REVIEW

File: client/src/pages/platform/workflow-editor/components/properties/hooks/useObjectProperty.ts [hunk 1]
```diff
@@ -316,7 +316,15 @@ export const useObjectProperty = ({onDeleteClick, path, property}: UseObjectProp
 
                 const parameterKeyValue = parameterObject[propertyDefinition.name];
 
-                if (parameterKeyValue !== undefined || objectParameterKeys.includes(propertyDefinition.name)) {
+                // Include properties if:
+                // 1. They have a parameter value, OR
+                // 2. They are in the ordered keys, OR
+                // 3. They have a displayCondition (visibility determined by Property component)
```
Comment (kresimir-coko) [inline client/src/pages/platform/workflow-editor/components/properties/hooks/useObjectProperty.ts]: ne želimo komentare, kod mora biti sam po sebi dokumentacija (čini mi se kako je sam uvjet dovoljno čitak, komentari ne dodaju nikakvu vrijednost u ovom slučaju)

File: client/src/pages/platform/workflow-editor/components/properties/hooks/useProperty.ts
Comment (kresimir-coko) [inline client/src/pages/platform/workflow-editor/components/properties/hooks/useProperty.ts:1]: Komentari su i ode samo višak
